services/storage/in_memory_storage.go

```diff
@@ -358,12 +359,97 @@ func (storage *InMemoryStorage) resetCommitStorage(blockHeight primitives.BlockH
 	storage.commitStorage[blockHeight] = views
 	return views
 }
+
 func (storage *InMemoryStorage) resetViewChangeStorage(blockHeight primitives.BlockHeight) map[primitives.View]map[MemberIdStr]*interfaces.ViewChangeMessage {
 	views := make(map[primitives.View]map[MemberIdStr]*interfaces.ViewChangeMessage)
 	storage.viewChangeStorage[blockHeight] = views
 	return views
 }
 
+func (storage *InMemoryStorage) GetMessagesLogs(blockHeight primitives.BlockHeight, view primitives.View) []interfaces.MemberMessagesLog {
```

Comment:
its a little odd that in_memory_storage formats your logs messages for you.. I realize you use non public members here but maybe the better fix is to make getter functions to all the data you need to format the log. this way you don't need to define a new struct "MessagesLog"..

why not do all this logic in some logger object? or at least some helper method of `logViewMessages`.

At least, rename this to "AsMessageLog" or something similar to indicate this changes the representation of the state.

Reply:
refactored